samples/instance-with-autoscaling-config.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2024 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// sample-metadata:
+//  title: Creates a instance with autoscaling config.
+//  usage: node instance-with-autoscaling-config.js <INSTANCE_ID> <PROJECT_ID>
+
+'use strict';
+
+function main(instanceId = 'my-instance', projectId = 'my-project-id') {
+  async function createInstanceWithAutoscalingConfig() {
+    // [START spanner_create_instance_with_autoscaling_config]
+    // Imports the Google Cloud client library
+    const {Spanner, protos} = require('@google-cloud/spanner');
+
+    /**
+     * TODO(developer): Uncomment the following lines before running the sample.
+     */
+    // const projectId = 'my-project-id';
+    // const instanceId = 'my-instance';
+
+    // Creates a client
+    const spanner = new Spanner({
+      projectId: projectId,
+    });
+
+    // Get the instance admin client
+    const instanceAdminClient = spanner.getInstanceAdminClient();
+
+    const autoscalingConfig =
+      protos.google.spanner.admin.instance.v1.AutoscalingConfig.create({
+        // Only one of minNodes/maxNodes or minProcessingUnits/maxProcessingUnits
+        // can be set. Both min and max need to be set and
+        // maxNodes/maxProcessingUnits can be at most 10X of
+        // minNodes/minProcessingUnits.
+        autoscalingLimits:
+          protos.google.spanner.admin.instance.v1.AutoscalingConfig.AutoscalingLimits.create(
+            {
+              minNodes: 1,
+              maxNodes: 2,
+            }
+          ),
+        // highPriorityCpuUtilizationPercent and storageUtilizationPercent are both
+        // percentages and must lie between 0 and 100.
+        autoscalingTargets:
+          protos.google.spanner.admin.instance.v1.AutoscalingConfig.AutoscalingTargets.create(
+            {
+              highPriorityCpuUtilizationPercent: 65,
+              storageUtilizationPercent: 95,
+            }
+          ),
+      });
+
+    // Creates a new instance with autoscaling configuration
+    try {
+      console.log(
+        `Creating instance ${instanceAdminClient.instancePath(
+          projectId,
+          instanceId
+        )}.`
+      );
+      const [operation] = await instanceAdminClient.createInstance({
```

Comment:
Lets add a comment above this
`When autoscaling_config is enabled, node_count and processing_units fields need not be specified.`
Ref: https://github.com/googleapis/java-spanner/blob/4905921313c08f84902d990dd6f716579b57464f/samples/snippets/src/main/java/com/example/spanner/CreateInstanceWithAutoscalingConfigExample.java#L52C7-L53C32

Reply:
Done